### ISSUE TITLE
Broker may publish meta events about wrong URIs

### ISSUE THREAD

**Issue by DZabavchik** [crossbario/crossbar#2131]
`Broker.detach()` may publish `wamp.subscription.on_delete` about a wrong URI (one that still has subscribers)

Local variable capture when `self._reactor.callLater(0, _publish, subscription)`  used from a loop is passing last value of those locals.

This needs to be refactored to pass local vars to `call_later` as arguments:

```self._reactor.callLater(0, _publish, subscription, was_subscribed, was_deleted, was_last_subscriber...)```
